git.md

```diff
@@ -1,2 +1,139 @@
-# Git
+These practices:
+
+* are aspirational but when they can't be followed it's nice to explain why;
+* document heuristics, as detailed instructions may never cover all scenarios;
+* apply to shared commits (maybe after cleaning commits locally).
+
+## tl;dr
+
+* Shared commits should not be changed.
+* Changes in each shared commit should be closely related.
+
+## 1. Preserve shared history
+
+Shared commits should never change. Some Git commands preserve the history, but
+others don't. For example, `git commit --amend` and `git rebase --interactive`
+alter the git history; they help to overwrite  a messy commit history with
+a cleaner one; you should use them only before sharing your work. To cleanup
```

Comment:
I'm thinking of including a glossary. That way the main text could keep short by using a succinct vocabulary we define elsewhere.

Reply:
that sounds excellent to me.... I guess it's not possible in markdown (?) but it would be cool if you could hover over the word to get its definition

Reply:
>  "sharing" also feels ambiguous to me... can we be more precise with this, like "pushed to origin" or something?

Thanks! The definition of "shared commit" now reads "for example, because you pushed it to a remote repository that
someone else has permission privileges to access".

> it would be cool if you could hover over the word to get its definition

The closest I could think of is a footnote, which I now prefer over the glossary because in the source code we can keep the definition close to the term being defined, and in the rendered output the term and its definition are just one click (not hover) away.